File: sdks/go/pkg/models/shared/zipdownloadstatus.go
package shared

type ZipDownloadStatusStateEnum string

const (
	ZipDownloadStatusStateEnumInProgress ZipDownloadStatusStateEnum = "in_progress"
	ZipDownloadStatusStateEnumFailed     ZipDownloadStatusStateEnum = "failed"
	ZipDownloadStatusStateEnumSuccess    ZipDownloadStatusStateEnum = "success"
)

// ZipDownloadStatus
// The status of a `zip` archive being downloaded.
type ZipDownloadStatus struct {
	DownloadedFileCount *int64                      `json:"downloaded_file_count,omitempty"`
	SkippedFileCount    *int64                      `json:"skipped_file_count,omitempty"`
	SkippedFolderCount  *int64                      `json:"skipped_folder_count,omitempty"`
	State               *ZipDownloadStatusStateEnum `json:"state,omitempty"`
	TotalFileCount      *int64                      `json:"total_file_count,omitempty"`
}

File: sdks/ts/src/sdk/models/shared/zipdownloadstatus.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum ZipDownloadStatusStateEnum {
    InProgress = "in_progress",
    Failed = "failed",
    Success = "success"
}


// ZipDownloadStatus
/** 
 * The status of a `zip` archive being downloaded.
**/
export class ZipDownloadStatus extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=downloaded_file_count" })
  downloadedFileCount?: number;

  @SpeakeasyMetadata({ data: "json, name=skipped_file_count" })
  skippedFileCount?: number;

  @SpeakeasyMetadata({ data: "json, name=skipped_folder_count" })
  skippedFolderCount?: number;

  @SpeakeasyMetadata({ data: "json, name=state" })
  state?: ZipDownloadStatusStateEnum;

  @SpeakeasyMetadata({ data: "json, name=total_file_count" })
  totalFileCount?: number;
}
